internal/plugin/process_monitor_plugin.go

```diff
@@ -105,22 +105,45 @@ func (pm *ProcessMonitor) run(ctx context.Context) {
 		case <-ctx.Done():
 			return
 		case <-pm.processTicker.C:
-			newCtx := context.WithValue(ctx, logger.CorrelationIDContextKey, logger.GenerateCorrelationID())
-			slog.DebugContext(newCtx, "Checking for process changes")
+			slog.DebugContext(ctx, "Checking for process updates")
 
-			processes, err := pm.getProcessesFunc(newCtx)
+			processes, err := pm.getProcessesFunc(ctx)
 			if err != nil {
-				slog.ErrorContext(newCtx, "Unable to get process information", "error", err)
+				slog.ErrorContext(ctx, "Unable to get process information", "error", err)
 
 				continue
 			}
 
 			pm.processesMutex.Lock()
-			pm.processes = processes
+			if haveProcessesChanged(pm.processes, processes) {
+				newCtx := context.WithValue(ctx, logger.CorrelationIDContextKey, logger.GenerateCorrelationID())
+				slog.DebugContext(newCtx, "Processes changes detected")
+				pm.processes = processes
+				pm.messagePipe.Process(newCtx, &bus.Message{Topic: bus.OsProcessesTopic, Data: processes})
+			}
 			pm.processesMutex.Unlock()
+		}
+	}
+}
+
+func haveProcessesChanged(oldProcesses, newProcesses []*model.Process) bool {
+	haveChanged := false
 
-			pm.messagePipe.Process(newCtx, &bus.Message{Topic: bus.OsProcessesTopic, Data: processes})
-			slog.DebugContext(newCtx, "Processes updated")
+	if len(oldProcesses) == len(newProcesses) {
+		processIDMap := make(map[int32]struct{})
+		for _, oldProcess := range oldProcesses {
+			processIDMap[oldProcess.Pid] = struct{}{}
 		}
+
+		for _, newProcess := range newProcesses {
+			if _, ok := processIDMap[newProcess.Pid]; !ok {
+				haveChanged = true
+				break
+			}
+		}
+	} else {
+		haveChanged = true
 	}
+
+	return haveChanged
 }
```

Comment:
Totally optional but I think something more like the following would be easier to read:
```go
func haveProcessesChanged(oldProcesses, newProcesses []*model.Process) bool {
	if len(oldProcesses) != len(newProcesses) {
		return true
	}

	oldPIDMap := make(map[int32]struct{})
	for _, oldProcess := range oldProcesses {
		oldPIDMap[oldProcess.Pid] = struct{}{}
	}

	for _, newProcess := range newProcesses {
		if _, ok := oldPIDMap[newProcess.Pid]; !ok {
			return true
		}
	}

	return false
}
```

Reply:
Updated and added some comments to make it easier to read.